$OpenBSD: patch-testsuite_mk_test_mk,v 1.10 2019/09/30 11:44:18 kili Exp $

Fix detection of available library ways in case the library-dirs:
field of ghc-prim contains more than one entry (only checks file
in the first entry, which is buggy, the whole detection mechanism
looks buggy).

Disable LLVM tests for now (until I unslack and debug why all LLVM
tests are failing).

Index: testsuite/mk/test.mk
--- testsuite/mk/test.mk.orig
+++ testsuite/mk/test.mk
@@ -93,7 +93,7 @@ else
 RUNTEST_OPTS += -e ghc_with_native_codegen=False
 endif
 
-GHC_PRIM_LIBDIR := $(subst library-dirs: ,,$(shell "$(GHC_PKG)" field ghc-prim library-dirs --simple-output))
+GHC_PRIM_LIBDIR := $(subst library-dirs: ,,$(shell "$(GHC_PKG)" field ghc-prim library-dirs --simple-output | sed -e 's/[[:space:]].*//'))
 HAVE_VANILLA := $(shell if [ -f $(subst \,/,$(GHC_PRIM_LIBDIR))/GHC/PrimopWrappers.hi ]; then echo YES; else echo NO; fi)
 HAVE_DYNAMIC := $(shell if [ -f $(subst \,/,$(GHC_PRIM_LIBDIR))/GHC/PrimopWrappers.dyn_hi ]; then echo YES; else echo NO; fi)
 HAVE_PROFILING := $(shell if [ -f $(subst \,/,$(GHC_PRIM_LIBDIR))/GHC/PrimopWrappers.p_hi ]; then echo YES; else echo NO; fi)
@@ -178,16 +178,7 @@ else
 RUNTEST_OPTS += -e ghc_with_smp=False
 endif
 
-ifeq "$(LLC)" ""
 RUNTEST_OPTS += -e ghc_with_llvm=False
-else ifeq "$(TargetARCH_CPP)" "powerpc"
-RUNTEST_OPTS += -e ghc_with_llvm=False
-else ifneq "$(LLC)" "llc"
-# If we have a real detected value for LLVM, then it really ought to work
-RUNTEST_OPTS += -e ghc_with_llvm=True
-else
-RUNTEST_OPTS += -e ghc_with_llvm=False
-endif
 
 ifeq "$(WINDOWS)" "YES"
 RUNTEST_OPTS += -e windows=True
